azure-iot-device/azure/iot/device/common/unhandled_exceptions.py

```diff
@@ -0,0 +1,27 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
```

Comment:
Also, I don't love that this is it's own module, but I see why it has to be - it's used in multiple places (`operation_flow` and `pipeline_stages_base`). This is an consequence of `operation_flow` being made into it's own module rather than having it's contents implemented as part of` pipelinge_stages_base`, the purpose of which I was never exactly clear on. 

I'm not suggesting any changes to that approach right now, but if we ever revisit and unify, we should probably fold this into the unification as well.

Reply:
I'm working off the assumption that this functionality will get at least slightly more complex and will get reworked.  This may be a temporary holding point.  I mostly wanted something simple that didn't make any assumptions.  For example, if we're returning these errors to the user somehow, this maybe needs to be exposed by the client object(s), but I'm not ready to take on the work to put it there until we're sure that's the right place  

(as far as operation_flow.py goes, we can discuss offline -- it's really more of a philosophical debate.)

---
In reply to: [305482114](https://github.com/Azure/azure-iot-sdk-python-preview/pull/133#discussion_r305482114) [](ancestors = 305482114)